src/fsharp/pars.fsy

```diff
@@ -1368,9 +1369,10 @@ attributeList:
       { if not $4 then reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedLBrackLess())
         $2 }
 
-  | LBRACK_LESS  ends_coming_soon_or_recover 
-      { if not $2 then reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedLBrackLess())
-        [] }
+  | LBRACK_LESS  opt_OBLOCKSEP ends_coming_soon_or_recover 
+      { if not $3 then reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedLBrackLess())
+        let rhsm = rhs parseState 1
+        [{ TypeName=Ast.LongIdentWithDots([], []); ArgExpr=(mkSynUnit rhsm); Target=None; AppliesToGetterAndSetter=false; Range=rhsm }] }
```

Comment:
I think this is the only place where we create a LongIdent that is empty.  It might be better to not break that (undocumented) invariant and instead use an option type here

Reply:
Originally, I wanted "option", but I found this documentation on LongIdentWithDots: https://github.com/Microsoft/visualfsharp/blob/master/src/fsharp/ast.fs#L163

```fsharp
/// LongIdent can be empty list - it is used to denote that name of some AST element is absent (i.e. empty type name in inherit)
```
When I saw that and then noticed that changing this to "option" would require a slew of changes to our codebase, it made sense to use an empty list.